solana/program_v2/programs/gateway_v2/src/state/network.rs

```diff
@@ -99,44 +99,25 @@ impl GatekeeperNetwork {
             > 0
     }
 
-    pub fn set_expire_time(
-        &mut self,
-        data: &UpdateNetworkData,
-        authority: &mut Signer,
-    ) -> Result<()> {
-        match data.pass_expire_time {
-            Some(pass_expire_time) => {
-                if pass_expire_time != self.pass_expire_time {
-                    if !self.can_access(authority, NetworkKeyFlags::SET_EXPIRE_TIME) {
-                        return Err(error!(NetworkErrors::InsufficientAccessExpiry));
-                    }
-
-                    self.pass_expire_time = pass_expire_time;
-                }
-
-                Ok(())
+    pub fn set_expire_time(&mut self, pass_expire_time: Option<i64>) -> Result<()> {
+        if let Some(pass_expire_time) = pass_expire_time {
+            if pass_expire_time != self.pass_expire_time {
```

Comment:
That if condition does not do anything, correct? Why not just assign?

Reply:
Yeah... the check for access what moved to a constraint so this is no longer needed

Reply:
Not sure I understand this comment? This line does not do access check, but just if old != new, correct? If I assign a new, why do I need to check that inequality here?

Reply:
yeah, i mean this check is now legacy/redundant since the access check was moved to a constraint.. i removed it